department-of-veterans-affairs/vets-api
# Bdex/57603: EVSS Migration: rated_disabilities - match app behavior response out side of successful response

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.


## Summary

- adds a `Lighthouse::ServiceException` handler that handles known errors from the Lighthouse APIs for the controller's `ExceptionHandling` class to consume
- Benefits Disability Experience
- no feature toggled added added

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#57603
- [*Link to ticket created in va.gov-team repo*](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/57603)


## Testing done

- a whole ton of unit and integration tests
- turn on the feature toggle `disability_compensation_lighthouse_rated_disabilities_provider`, create a new `LighthouseRatedDisabilitiesProvider` and call `get_rated_disabilities` with a client_id and an rsa_key corresponding to it. 
- use an ICN that Lighthouse does not have in their database to get a 404 error and see that the `LighthouseRatedDisabilitiesProvider` has the error "Veteran not identifiable." in the response body

## Screenshots
![image](https://user-images.githubusercontent.com/19178435/236336489-1d5a6349-846d-45c6-82f6-21c3accd6c03.png)

## What areas of the site does it impact?
- rated_disabilities if the feature toggle `disability_compensation_lighthouse_rated_disabilities_provider` is toggled on

## Acceptance criteria


## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
